repos/system_upgrade/el8toel9/actors/firewalldcheckallowzonedrifting/actor.py

```diff
@@ -0,0 +1,51 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import FirewalldGlobalConfig, FirewallsFacts
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class FirewalldCheckAllowZoneDrifting(Actor):
+    """
+    This actor will check if AllowZoneDrifiting=yes in firewalld.conf. This
+    option has been removed in RHEL-9 and behavior is as if
+    AllowZoneDrifiting=no.
+    """
+
+    name = 'firewalld_check_allow_zone_drifting'
+    consumes = (FirewallsFacts, FirewalldGlobalConfig)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        # If firewalld is not enabled then don't bother the user about its
+        # configuration. This Report keys off a _default_ value and as such
+        # will trigger for all users that have not done one of the following:
+        #   - disabled firewalld
+        #   - manually set AllowZoneDrifting=no (as firewalld logs suggests)
+        #
+        for facts in self.consume(FirewallsFacts):
+            if not facts.firewalld.enabled:
+                return
+
+        for facts in self.consume(FirewalldGlobalConfig):
+            if not facts.allowzonedrifting:
+                return
+
+        create_report([
+            reporting.Title('Unsupported Firewalld Configuration In Use'),
+            reporting.Summary('Firewalld has enabled configuration option '
+                              '"{conf_key}" which has been removed in RHEL-9. '
+                              'New behavior is as if "{conf_key}" was set to "no".'.format(
+                                  conf_key='AllowZoneDrifiting')),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SANITY, reporting.Tags.FIREWALL]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.ExternalLink(
+                url='https://access.redhat.com/articles/4855631',
+                title='Changes in firewalld related to Zone Drifting'),
+            reporting.Remediation(
+                hint='Set AllowZoneDrifting=no in /etc/firewalld/firewalld.conf',
+                commands=[['sh', '-c',
```

Comment:
As this is just a single command, please remove the "sh -c" and let's go with the sed command only. 

Now, in the report, it translates to:
`sh -c sed -i "s/^AllowZoneDrifting=.*/AllowZoneDrifting=no/" /etc/firewalld/firewalld.conf` which is broken without apostrophes around the sed command.
Thanks...

Reply:
Oh. I thought this was doing shell escaping. I followed the `sh -c` pattern from this example: https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/actors/checkhacluster/libraries/checkhacluster.py

I'll change it to just call `sed` directly.

Is there a way to test this via snactor?

Reply:
I do not think we can test it via snactor but you can test it using Leapp cockpit plugin. I can share details with you offline.

Reply:
In any way, when the command is in a good shape and executable, we should be good to merge this.

Reply:
done.